publish-counts.js

```diff
@@ -33,12 +33,62 @@ if (Meteor.isServer) {
     if (countFn && options.nonReactive)
       throw new Error("options.nonReactive is not yet supported with options.countFromFieldLength or options.countFromFieldSum");
 
-    if ('function' !== typeof extraField) {
-      // ensure the cursor doesn't fetch more than it has to
-      cursor._cursorDescription.options.fields = {_id: true};
-      if (extraField)
-        cursor._cursorDescription.options.fields[extraField] = true;
-    }
+    // begin cursor field limit optimizations
+
+    // ensure the cursor doesn't fetch more fields than it has to.
+    // keep user defined field limits except with basic count.
+    cursor._cursorDescription.options.fields = (function (fields, extraField) {
```

Comment:
Certainly.  I like functions and figured this block might later migrate to one, which is part of the reason I used an IIFE in this case.  I just wasn't sure where to put it.